airflow/contrib/hooks/gcp_dataflow_hook.py

```diff
@@ -124,36 +127,48 @@ def __init__(self, cmd):
 
     def _line(self, fd):
         if fd == self._proc.stderr.fileno():
-            lines = self._proc.stderr.readlines()
-            for line in lines:
-                self.log.warning(line[:-1])
-            if lines:
-                return lines[-1]
+            line = ''.join(self._proc.stderr.readlines())
+            self.log.warning(line[:-1])
+            return line
         if fd == self._proc.stdout.fileno():
-            line = self._proc.stdout.readline()
+            line = ''.join(self._proc.stdout.readlines())
+            self.log.info(line[:-1])
             return line
 
     @staticmethod
     def _extract_job(line):
-        if line is not None:
-            if line.startswith("Submitted job: "):
-                return line[15:-1]
+        # Job id info: https://goo.gl/SE29y9.
+        job_id_pattern = re.compile(
+            b'.*console.cloud.google.com/dataflow.*/jobs/([a-z|0-9|A-Z|\-|\_]+).*')
```

Comment:
Can you also include location in this regex e.g. '.../dataflow/jobsDetail/locations/${location}/jobs/'? In case a job is running in multiple locations.

Reply:
Note that this is the URL of the Cloud Console job monitoring page where job-id is a flat thing (i.e., no location hierarchy).

Reply:
The linked line includes location- I think location is in the hierarchy from looking at it as well.

Reply:
Actually this should be fine. Even though there is a location in the hierarchy, the operator is only launching one job, so job id collision is not a concern.